dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/importer/EventManager.java

```diff
@@ -293,6 +324,57 @@ public ImportSummaries deleteEvents( final List<Event> events, final WorkContext
         return importSummaries;
     }
 
+    private void logTrackedEntityDataValueHistory( Event event, WorkContext workContext, Date today )
+    {
+        String persistedDataValue = null;
```

Comment:
Please declare vars in the smallest possible/necessary scope. If there are no event.dataValues you don't even need the var.

Reply:
agreed. But if I move it to the smallest possible scope then that ends up in a loop and I don't need to create this var in each loop iteration.

Reply:
If you are certain this is hurting performance then keep it the way it is 😋 otherwise I would opt for readability/correctness first.